gitlabform/processors/project/hooks_processor.py

```diff
@@ -9,23 +13,31 @@ def __init__(self, gitlab: GitLab):
         super().__init__("hooks", gitlab)
 
     def _process_configuration(self, project_and_group: str, configuration: dict):
+        debug("Processing hooks...")
+        project: Project = self.gl.projects.get(project_and_group)
+        hooks_list: RESTObjectList | List[RESTObject] = project.hooks.list()
+
         for hook in sorted(configuration["hooks"]):
+            gitlab_hook: RESTObject | None = next(
+                (h for h in hooks_list if h.url == hook), None
+            )
+            hook_id = gitlab_hook.id if gitlab_hook else None
             if configuration.get("hooks|" + hook + "|delete"):
-                hook_id = self.gitlab.get_hook_id(project_and_group, hook)
                 if hook_id:
                     debug("Deleting hook '%s'", hook)
-                    self.gitlab.delete_hook(project_and_group, hook_id)
+                    project.hooks.delete(hook_id)
                 else:
                     debug("Not deleting hook '%s', because it doesn't exist", hook)
             else:
-                hook_id = self.gitlab.get_hook_id(project_and_group, hook)
-                if hook_id:
-                    debug("Changing existing hook '%s'", hook)
-                    self.gitlab.put_hook(
-                        project_and_group, hook_id, hook, configuration["hooks"][hook]
-                    )
-                else:
+                hook_config = {"url": hook}
+                hook_config.update(configuration["hooks"][hook])
+                if not hook_id:
                     debug("Creating hook '%s'", hook)
-                    self.gitlab.post_hook(
-                        project_and_group, hook, configuration["hooks"][hook]
+                    created_hook: RESTObject = project.hooks.create(hook_config)
+                    debug("Created hook '%s'", created_hook)
+                elif gitlab_hook and gitlab_hook.asdict() != hook_config:
```

Comment:
Hi @TigreModerata!

Thank you for the contribution. :)

However I suspected that this condition will always be true because in `hook_config` we have only a few fields from a hook that we get from the API (`gitlab_hook.asdict()`), f.e. we don't have the field `id`.

And it turned out to be true - I modified a test `test_hooks_update` to do the update 2 times and in the second time the update is done again.

When I runned it with `pytest tests/acceptance/standard/test_hooks.py -k 'test_hooks_create or test_hooks_update'`.

First time logs excerpt:

```
Processing hooks...
http://localhost:80 "GET /api/v4/projects/gitlabform_group_papaya%2Fgitlabform_project_stony HTTP/1.1" 200 None
http://localhost:80 "GET /api/v4/projects/8/hooks HTTP/1.1" 200 None
Changing existing hook 'http://hooks/gitlabform_hook_festival.org'
http://localhost:80 "PUT /api/v4/projects/8/hooks/9 HTTP/1.1" 200 None
Changed hook to '{'id': 9, 'url': 'http://hooks/gitlabform_hook_festival.org', 'created_at': '2023-12-03T11:41:51.849Z', 'push_events': True, 'tag_push_events': False, 'merge_requests_events': False, 'repository_update_events': False, 'enable_ssl_verification': True, 'alert_status': 'executable', 'disabled_until': None, 'url_variables': [], 'project_id': 8, 'issues_events': False, 'confidential_issues_events': False, 'note_events': True, 'confidential_note_events': None, 'pipeline_events': False, 'wiki_page_events': False, 'deployment_events': False, 'job_events': True, 'releases_events': False, 'push_events_branch_filter': None, 'emoji_events': False}'
Changing existing hook 'http://hooks/gitlabform_hook_tiring.org'
http://localhost:80 "PUT /api/v4/projects/8/hooks/10 HTTP/1.1" 200 None
Changed hook to '{'id': 10, 'url': 'http://hooks/gitlabform_hook_tiring.org', 'created_at': '2023-12-03T11:41:51.907Z', 'push_events': False, 'tag_push_events': False, 'merge_requests_events': False, 'repository_update_events': False, 'enable_ssl_verification': True, 'alert_status': 'executable', 'disabled_until': None, 'url_variables': [], 'project_id': 8, 'issues_events': False, 'confidential_issues_events': False, 'note_events': True, 'confidential_note_events': None, 'pipeline_events': False, 'wiki_page_events': False, 'deployment_events': False, 'job_events': False, 'releases_events': False, 'push_events_branch_filter': None, 'emoji_events': False}'
* (1/1) FINISHED Processing project: gitlabform_group_papaya/gitlabform_project_stony
```

Second time logs excerpt:
```
Processing hooks...
http://localhost:80 "GET /api/v4/projects/gitlabform_group_papaya%2Fgitlabform_project_stony HTTP/1.1" 200 None
http://localhost:80 "GET /api/v4/projects/8/hooks HTTP/1.1" 200 None
Changing existing hook 'http://hooks/gitlabform_hook_festival.org'
http://localhost:80 "PUT /api/v4/projects/8/hooks/9 HTTP/1.1" 200 None
Changed hook to '{'id': 9, 'url': 'http://hooks/gitlabform_hook_festival.org', 'created_at': '2023-12-03T11:41:51.849Z', 'push_events': True, 'tag_push_events': False, 'merge_requests_events': False, 'repository_update_events': False, 'enable_ssl_verification': True, 'alert_status': 'executable', 'disabled_until': None, 'url_variables': [], 'project_id': 8, 'issues_events': False, 'confidential_issues_events': False, 'note_events': True, 'confidential_note_events': None, 'pipeline_events': False, 'wiki_page_events': False, 'deployment_events': False, 'job_events': True, 'releases_events': False, 'push_events_branch_filter': None, 'emoji_events': False}'
Changing existing hook 'http://hooks/gitlabform_hook_tiring.org'
http://localhost:80 "PUT /api/v4/projects/8/hooks/10 HTTP/1.1" 200 None
Changed hook to '{'id': 10, 'url': 'http://hooks/gitlabform_hook_tiring.org', 'created_at': '2023-12-03T11:41:51.907Z', 'push_events': False, 'tag_push_events': False, 'merge_requests_events': False, 'repository_update_events': False, 'enable_ssl_verification': True, 'alert_status': 'executable', 'disabled_until': None, 'url_variables': [], 'project_id': 8, 'issues_events': False, 'confidential_issues_events': False, 'note_events': True, 'confidential_note_events': None, 'pipeline_events': False, 'wiki_page_events': False, 'deployment_events': False, 'job_events': False, 'releases_events': False, 'push_events_branch_filter': None, 'emoji_events': False}'
* (1/1) FINISHED Processing project: gitlabform_group_papaya/gitlabform_project_stony
```

This is not great as it lowers the performance of a run and may cause unnecessary audit events in the projects on each run ("Hook ... has been updated").

I have pushed my commit to show the problem here https://github.com/gitlabform/gitlabform/commit/d8b84e1a4aa5860564059a8a829e916a781ea9a7.

Reply:
@TigreModerata - no rush but just wondering if you will be able to look into this feedback? I feel this PR is so close to being merged🤞 

It's a good callout from @gdubicki . I missed this.